### ISSUE TITLE
Feature: transfers of intensive variables conserving a different extensive quantity

### ISSUE THREAD

**Issue by GiudGiud** [idaholab/moose#16428]
## Reason
Every time we work with an intensive variable and transfer it to another app, we have the option to conserve its integral over the domain. However, for intensive variables like temperature, conserving the integral of temperature makes no sense, it's conserving the extensive quantity (energy) that is desirable. 

Other example: volume averaged neutron fluxes and reaction rates

## Design
The transfer should have the option to conserve the extensive quantity over the whole mesh, over each block or even over each element if the same mesh is used. 
The link between the intensive variables and the conservative one may be done through material properties, postprocessors and parsed functions. 

## Impact
This will give us many physics karma points for conserving energy as often as we can. 
This will enable multi-scale approaches for heat conduction solves, with large mesh full domain solves coupled to targeted fine mesh solves.

**Comment by fdkong** [comment]
The current conservative transfer already supports this

**Comment by GiudGiud** [comment]
It only supports preserving a postprocessor. I'm asking for a lot more capability here.

With block-restricted transfers you could create a conservative-by-block transfer, but that could be a lot of transfers to create depending on the number of blocks. 

How about trying to preserve a VectorPostprocessor, with the vector being over blocks?

**Comment by fdkong** [comment]
> With block-restricted transfers you could create a conservative-by-block transfer, but that could be a lot of transfers to create depending on the number of blocks.

Could you post what the input would look like?

**Comment by GiudGiud** [comment]
Conservative transfers one block at a time:

```
[Transfers]
  [./to_sub]
    type = MultiAppMeshFunctionTransfer
    direction = to_multiapp
    source_variable = u
    variable = aux_u
    multi_app = sub
    from_postprocessors_to_be_preserved = 'from_postprocessor' #assumed to only work on block 1
    to_postprocessors_to_be_preserved = 'to_postprocessor'         # assumed to only work on block 1
    block = 1
  [../]
[]
[Transfers]
  [./to_sub]
    type = MultiAppMeshFunctionTransfer
    direction = to_multiapp
    source_variable = u
    variable = aux_u
    multi_app = sub
    from_postprocessors_to_be_preserved = 'from_postprocessor'
    to_postprocessors_to_be_preserved = 'to_postprocessor'
    block = 2
  [../]

  # Same for block 3, 4, 5 ...
[]
```

One transfer, conserving on every block:
```
[Transfers]
  [./to_sub]
    type = MultiAppMeshFunctionTransfer
    direction = to_multiapp
    source_variable = u
    variable = aux_u
    multi_app = sub
    from_postprocessors_to_be_preserved = 'from_vector_postprocessor_by_block'
    to_postprocessors_to_be_preserved = 'to_vector_postprocessor_by_block'
  [../]
[]
```

**Comment by fdkong** [comment]
OK, we need to figure what blocks we are going to use or we can supply parameters like

```
to_blocks = "1 2 3 4"
from_blocs = " 4 5 1 3"
```

**Comment by GiudGiud** [comment]
Yes that would help option 2 I think. Though maybe there should be a way to read the vector postprocessors's blocks. 

Because if the first entry in the postprocessor is the 2nd block transfered it could get awkward

**Comment by fdkong** [comment]
> Though maybe there should be a way to read the vector postprocessors's blocks.

It is a better option to me
